### PR TITLE
Fix eos.Parameter.FromWCxf

### PR DESCRIPTION
**Pull request by dvandyk**
- depends on merging wilson-eft/wilson#71
 - add test case

### CODE REVIEW

File: python/eos/parameter.py [hunk 1]
```diff
@@ -99,24 +99,40 @@ def _repr_html_(self):
...
+
+        # extract wc data
+        wc = w.wc
+
+        if wc.basis != 'EOS':
+            raise ValueError('Wilson coefficients must be passed to this function in the EOS basis')
+
+        # Needs to be revisited, once EOS support WET-4 or WET-3 bases
+        if wc.scale != 4.2:
+            raise ValueError('Wilson coefficients must be passed to this function at the reference scale 4.2 GeV')
```
Comment (mreboud) [inline python/eos/parameter.py:119]: Should we open an issue to do this automatically?

File: python/eos/parameter.py [hunk 2]
```diff
@@ -99,24 +99,40 @@ def _repr_html_(self):
...
         parameters = _Parameters.Defaults()
-        for name in wc.dict:
-            qn     = eos.QualifiedName(name)
+        for name, value in wc.dict.items():
+            qn     = QualifiedName(name)
             prefix = qn.prefix_part()
             coeff  = qn.name_part()
-            value  = wc.dict[name]
-            if (not value.imag == 0) and (coeff in real_coeffs):
-                raise ValueError('WC {0} does not support non-zero imaginary part'.format(name))
```
Comment (mreboud) [inline python/eos/parameter.py:116]: I remember that Peter Stangl mentioned small imaginary part due to running and matching https://github.com/eos/eos/issues/335#issuecomment-874747524

File: python/eos/parameter.py [hunk 3]
```diff
@@ -99,24 +99,40 @@ def _repr_html_(self):
...
 
             # Add values provided by WCxf to EOS central (SM) values
-            if str(coeff) in real_coeffs:
+            if (prefix == 'b->s') and (coeff in real_coeffs):
+                if (abs(value.imag) < 1.0e-10):
```
Comment (mreboud) [inline python/eos/parameter.py:133]: Shouldn't this be `> 1e-10`?

File: python/eos/parameter_TEST.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+class ClassMethodTests(unittest.TestCase):
+
+    def test_importing_from_wcxf(self):
+
+        from wilson import Wilson
+
+        inputs = {
+	    'cbtaunutau::cSL':  0.3 + 0.1j,
+            'ubenue::cVL':     -0.1 + 0.2j,
```
Comment (mreboud) [inline python/eos/parameter_TEST.py:14]: Indentations look weird, no?

File: python/eos/parameter_TEST.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+        }
+        w = Wilson(inputs, scale=4.2, eft="WET", basis="EOS")
+        p = eos.Parameters.FromWCxf(w)
+
```
Comment (mreboud) [inline python/eos/parameter_TEST.py:18]: Can we check the values of `p`, to ensure correct import? This is more for future updates of the code...

**Review by mreboud — CHANGES_REQUESTED** [review]

File: python/eos/parameter.py (hunk 3 above)
Comment (dvandyk) [reply python/eos/parameter.py:133]: And this is why a test case is needed!

File: python/eos/parameter_TEST.py (hunk 4 above)
Comment (dvandyk) [reply python/eos/parameter_TEST.py:14]: Yes, will fix it.

File: python/eos/parameter_TEST.py (hunk 5 above)
Comment (dvandyk) [reply python/eos/parameter_TEST.py:18]: Yes. To be kept in mind: Wilson/WCxf parametrizes the BSM contributions. We still have to add the SM contributions on top of that.

File: python/eos/parameter.py (hunk 1 above)
Comment (dvandyk) [reply python/eos/parameter.py:119]: Hmm, I kind of like that the user must be aware of this issue. It should not be hardship for the user to use match_run correctly.

File: python/eos/parameter.py (hunk 2 above)
Comment (dvandyk) [reply python/eos/parameter.py:116]: Indeed, see below.

File: python/eos/parameter.py (hunk 3 above)
Comment (dvandyk) [reply python/eos/parameter.py]: Added to the test case.

**Comment by dvandyk** [comment]
I am now updating the CI/CD images to include wilson.

**Review by mreboud — APPROVED** [review]

**Pull request merged**